src/test/rpc_wallet_tests.cpp

```diff
@@ -1520,21 +1520,21 @@ BOOST_AUTO_TEST_CASE(rpc_z_shieldcoinbase_parameters)
     std::string mainnetzaddr = "zcMuhvq8sEkHALuSU2i4NbNQxshSAYrpCExec45ZjtivYPbuiFPwk6WHy4SvsbeZ4siy1WheuRGjtaJmoD1J8bFqNXhsG6U";
 
     try {
-        std::shared_ptr<AsyncRPCOperation> operation(new AsyncRPCOperation_shieldcoinbase(mtx, {}, testnetzaddr, -1 ));
+        std::shared_ptr<AsyncRPCOperation> operation(new AsyncRPCOperation_shieldcoinbase(TransactionBuilder(), mtx, {}, testnetzaddr, -1 ));
     } catch (const UniValue& objError) {
         BOOST_CHECK( find_error(objError, "Fee is out of range"));
     }
 
     try {
-        std::shared_ptr<AsyncRPCOperation> operation(new AsyncRPCOperation_shieldcoinbase(mtx, {}, testnetzaddr, 1));
+        std::shared_ptr<AsyncRPCOperation> operation(new AsyncRPCOperation_shieldcoinbase(TransactionBuilder(), mtx, {}, testnetzaddr, 1));
     } catch (const UniValue& objError) {
         BOOST_CHECK( find_error(objError, "Empty inputs"));
     }
 
     // Testnet payment addresses begin with 'zt'.  This test detects an incorrect prefix.
```

Comment:
While the first sentence is technically correct for Sapling testnet shielded addresses (which begin with "ztestsapling"), I'm skeptical that this actually tests what it claims, since the code here doesn't mention "zt".

Reply:
This is still a Sprout test; see above comment.

Reply:
I'm still skeptical that it tests what it claims, but OK, that means it's not a new issue for this PR. Maybe file a follow-up ticket.

Reply:
The test uses `SelectParams(CBaseChainParams::TESTNET)` so it requires testnet addresses; the test below uses a mainnet Sprout address. This is effectively checking that `DecodePaymentAddress()` is being called correctly and having its errors returned.

Reply:
Right, but the comment implies that the test would detect if a prefix not starting with "zt" were used, which can't be the case because the test doesn't mention "zt".

Reply:
The test *would* detect this, because the address gets passed through to `DecodePaymentAddress()`, which enforces the `zt` prefix. The fact we're only passing in an address with a `zc` prefix doesn't change the fact that any other prefix would also fail.